chat/src/components/message/BotIsTypingMessage.tsx

```diff
@@ -0,0 +1,74 @@
+import { useFetchBotResponse } from "../../hooks/useFetchBotResponse";
+import { ChatMessages, MessageResponse } from "../../types/ChatTypes";
+import { ChatBotQueryData } from "../../hooks/useChatBotQueryData";
+import { useEffect } from "react";
+import styled from "styled-components";
+import { AbstractMessage } from "./AbstractMessage";
+import { MessageContextProvider } from "../../hooks/useMessageContext";
+import { useConversationContext } from "../../hooks/useConversationContext";
+
+type Props = {
+  chatMessages: ChatMessages;
+  chatBotQueryData: ChatBotQueryData;
+  onTextChange(response: MessageResponse): void;
+  onFinish(response: MessageResponse): void;
+  onError(response: MessageResponse): void;
+};
+
+export function BotIsTypingMessage({
+  chatMessages,
+  chatBotQueryData,
+  onFinish,
+  onError,
+  onTextChange,
+}: Props): React.ReactElement | null {
+  const { id: conversationId } = useConversationContext();
+  const { data, isLoading, error } = useFetchBotResponse(
+    chatMessages,
+    chatBotQueryData
+  );
+  const { messageId } = chatBotQueryData;
+
+  useEffect(() => {
+    onTextChange({
+      id: messageId,
+      content: data,
+    });
+    if (error) {
+      onError({
+        id: messageId,
+        content: error,
+      });
+      return;
+    }
+    if (!isLoading) {
+      onFinish({
+        id: messageId,
+        content: data,
+      });
+      return;
+    }
+  }, [messageId, data, isLoading, error, onTextChange, onError, onFinish]);
```

Comment:
Feels like every part here deserves it's own `useEffect` (with dedicated dep array)